javascript/javascript3/week3/index.js

```diff
@@ -0,0 +1,38 @@
+// Paint a circle o a canvas element
+
+// Class creation time!
+class Circle {
+  constructor(x, y, r, startAngle, endAngle, fillColor) {
+    this.x = x;
+    this.y = y;
+    this.r = r;
+    this.startAngle = startAngle;
+    this.endAngle = endAngle;
+    this.fillColor = fillColor;
+  }
+  draw() {
+    const canvas = document.getElementById("circle");
+    canvas.style.width = "100%";
+    canvas.style.height = "100%";
+    const ctx = canvas.getContext("2d");
+    ctx.beginPath();
+    ctx.arc(this.x, this.y, this.r, this.startAngle, this.endAngle);
+    ctx.fillStyle = this.fillColor;
+    ctx.stroke();
+    ctx.fill();
```

Comment:
To make the class reusable, we can pass "context" as an argument to the draw method.
 
 const canvas = document.getElementById("circle");
    canvas.style.width = "100%";
    canvas.style.height = "100%";
    const ctx = canvas.getContext("2d");

draw(context) {
    ctx.beginPath();
    ctx.arc(this.x, this.y, this.r, this.startAngle, this.endAngle);
    ctx.fillStyle = this.fillColor;
    ctx.stroke();
    ctx.fill();
}

const c1 = new Circle(50, 50, 20, 0, 2 * Math.PI, "yellow");
c1.draw(ctx);

Reply:
@ElhamDargi Hi. Thank you for review. Please, take a look , I changed this part.